units/host_termination.go

```diff
@@ -218,13 +219,15 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
 					return
 				}
 				// Check for the last task in the task group that we have activated, running, or completed.
-				var lastTaskGroupTask task.Task
+				var lastActivatedTaskGroupTask task.Task
 				for _, t := range tasks {
 					if t.Activated {
-						lastTaskGroupTask = t
+						lastActivatedTaskGroupTask = t
 					}
 				}
-				if lastTaskGroupTask.Id != latestTask.Id {
+				// We want to block all subsequent tasks from running, regardless of what status the dependencies are waiting on.
```

Comment:
Can we explain why in this specific case we'd want to block dependencies of the single host task group task regardless of status? Would help to explain the behavior that it should restart the task group from scratch but single host task groups will only restart when the task group is finished, and in this case it's not finished. Blocking dependent tasks is a workaround to force the task group to restart immediately.

Reply:
Added a more descriptive comment